openedx_learning/apps/authoring/components/models.py

```diff
@@ -120,14 +122,12 @@ class Component(PublishableEntityMixin):  # type: ignore[django-manager-missing]
     Make a foreign key to the Component model when you need a stable reference
     that will exist for as long as the LearningPackage itself exists.
     """
-    # Tell mypy what type our objects manager has.
-    # It's actually PublishableEntityMixinManager, but that has the exact same
-    # interface as the base manager class.
-    objects: models.Manager[Component] = WithRelationsManager(
+    # Set up our custom manager. It has the same API as the default one, but selects related objects by default.
+    objects: ClassVar[WithRelationsManager[Component]] = WithRelationsManager(  # type: ignore[assignment]
```

Comment:
Unfortunately this line still requires specifying the type explicitly and also needs `ignore[assignment]` because mypy for some reason thinks that `WithRelationsManager[Component]` and `WithRelationsManager[Self]` are different things in this case. Seems like a bug 🤷🏻‍♂️.

In all other cases, where we're just using `WithRelationsManager` directly or just relying on a mixin to provide objects, no type annotations or suppression are needed. It's just here where the override in `Component` seems to conflict with the override in `PublishableEntityMixin` (though it shouldn't).